jetbrains-core/src/software/aws/toolkits/jetbrains/services/caws/envclient/CawsEnvironmentClient.kt

```diff
@@ -88,6 +91,27 @@ class CawsEnvironmentClient(
         return objectMapper.readValue(response.entity.content)
     }
 
+    fun getActivity(): GetActivityResponse = try {
+        val request = HttpGet("$endpoint/activity")
+        val response = execute(request)
+        objectMapper.readValue<GetActivityResponse>(response.entity.content)
+    } catch (e: Exception) {
+        LOG.error(e) { "Couldn't parse response from /activity API" }
+        GetActivityResponse(timestamp = null)
```

Comment:
if we can't parse the response, we might not want to run any heartbeat logic?
one case where we can't parse is when the environment is not in heartbeat timeout mode

Reply:
Returning a null here in case we cant parse a response which is then handled in the main code